DigitalLearningSolutions.Web.Tests/Controllers/LearningMenu/LearningMenuControllerTests.cs

```diff
@@ -1122,6 +1125,218 @@ public void TutorialVideo_should_return_404_if_unable_to_enrol()
+            const int progressId = 101;
+
+            A.CallTo(() => courseCompletionService.GetCourseCompletion(CandidateId, CustomisationId))
+                .Returns(expectedCourseCompletion);
+            A.CallTo(() => courseContentService.GetOrCreateProgressId(CandidateId, CustomisationId, CentreId))
+                .Returns(progressId);
+
+            // When
+            var result = controller.CompletionSummary(CustomisationId);
+
+            // Then
+            var expectedModel = new CourseCompletionViewModel(expectedCourseCompletion);
+
+            result.Should().BeViewResult()
+                .Model.Should().BeEquivalentTo(expectedModel);
+        }
+
+        [Test]
+        public void CompletionSummary_should_return_404_if_invalid_course()
+        {
+            // Given
+            const int progressId = 101;
+
+            A.CallTo(() => courseCompletionService.GetCourseCompletion(CandidateId, CustomisationId))
+                .Returns(null);
+            A.CallTo(() => courseContentService.GetOrCreateProgressId(CandidateId, CustomisationId, CentreId))
+                .Returns(progressId);
+
+            // When
+            var result = controller.CompletionSummary(CustomisationId);
+
+            // Then
+            result.Should()
+                .BeRedirectToActionResult()
+                .WithControllerName("LearningSolutions")
+                .WithActionName("StatusCode")
+                .WithRouteValue("code", 404);
+        }
+
+        [Test]
+        public void CompletionSummary_should_return_404_if_unable_to_enrol()
+        {
+            // Given
+            var expectedCourseCompletion = CourseCompletionHelper.CreateDefaultCourseCompletion(CustomisationId);
+
+            A.CallTo(() => courseCompletionService.GetCourseCompletion(CandidateId, CustomisationId))
+                .Returns(expectedCourseCompletion);
+            A.CallTo(() => courseContentService.GetOrCreateProgressId(CandidateId, CustomisationId, CentreId))
+                .Returns(null);
+
+            // When
+            var result = controller.CompletionSummary(CustomisationId);
+
+            // Then
+            result.Should()
+                .BeRedirectToActionResult()
+                .WithControllerName("LearningSolutions")
+                .WithActionName("StatusCode")
+                .WithRouteValue("code", 404);
+        }
+
+        [Test]
+        public void CompletionSummary_should_return_404_if_course_does_not_include_certification()
+        {
+            // Given
+            var expectedCourseCompletion = CourseCompletionHelper.CreateDefaultCourseCompletion(
+                CustomisationId,
+                includeCertification: false
+            );
+            const int progressId = 101;
+
+            A.CallTo(() => courseCompletionService.GetCourseCompletion(CandidateId, CustomisationId))

```

Comment:
It might be more consistent if `GetCourseCompletion` returns null in this case? 🙂

Reply:
I agree but this would break the tests I currently have, because there are no courses in the test DB that have `IncludeCertification` as true. Either we need to change the data in the test DB, or we could leave these checks where they are in the learning menu controller. What do you think? 🙂

Reply:
We have a few scenarios like this already in the tests, where we want to test something we don't have the correct data for. In that case we've added a helper method to change the test db and just put that inside transaction scope in the test. We could also change the test db permanently but that would be a bit more hassle since we'd all need to do it on our copy of it and it means restoring the test db if it got into a bad state (which has happened a few times already) would be more hassle 🙂

Reply:
I've added a method to add certification to the test records (in a transaction scope), and moved this check `IncludeCertification` into the data service, in 1c742e1 🙂